src/crimson/admin/osd_admin.cc

```diff
@@ -93,6 +93,80 @@ class SendBeaconHook : public AdminSocketHook {
 template std::unique_ptr<AdminSocketHook>
 make_asok_hook<SendBeaconHook>(crimson::osd::OSD& osd);
 
+/**
+ * An OSD admin hook: run bench
+ * Usage parameters:
+ *   count=Count of bytes to write
+ *   bsize=block size
+ *   osize=Object size
+ *   onum=Number of objects
+ */
+class RunOSDBenchHook : public AdminSocketHook {
+public:
+  explicit RunOSDBenchHook(crimson::osd::OSD& osd) :
+    AdminSocketHook{"bench",
+        "name=count,type=CephInt,req=false "
+        "name=size,type=CephInt,req=false "
+        "name=object_size,type=CephInt,req=false "
+        "name=object_num,type=CephInt,req=false",
+        "run OSD bench"},
+    osd(osd)
+  {}
+  seastar::future<tell_result_t> call(const cmdmap_t& cmdmap,
+              std::string_view format,
+              ceph::bufferlist&& input) const final
+  {
+    int64_t count = cmd_getval_or<int64_t>(cmdmap, "count", 1LL << 30);
+    int64_t bsize = cmd_getval_or<int64_t>(cmdmap, "size", 4LL << 20);
+    int64_t osize = cmd_getval_or<int64_t>(cmdmap, "object_size", 0);
+    int64_t onum = cmd_getval_or<int64_t>(cmdmap, "object_num", 0);
+    double elapsed = 0.0;
+    auto duration = local_conf()->osd_bench_duration;
+
+    if (bsize > local_conf()->osd_bench_max_block_size) {
+      logger().info("bench block size {} > osd_bench_max_block_size {}",
+                    bsize, local_conf()->osd_bench_max_block_size);
+      return seastar::make_ready_future<tell_result_t>(-EINVAL, "block size too large");
+    } else if (bsize < (1LL << 20)) {
+      int64_t max_count = bsize * duration * local_conf()->osd_bench_small_size_max_iops;
+      if (count > max_count) {
+        logger().info("bench count {} > osd_bench_small_size_max_iops {}",
+                      count, max_count);
+        return seastar::make_ready_future<tell_result_t>(-EINVAL, "count too large");
+      }
+    } else {
+      int64_t max_count = local_conf()->osd_bench_large_size_max_throughput * duration;
+      if (count > max_count) {
+        logger().info("bench count {} > osd_bench_large_size_max_throughput {}, assuming {}/s,"
+                       " for {} seconds, can cause ill effects on osd.",
```

Comment:
`Please adjust 'osd_bench_small_size_max_iops' with a higher value if you wish to use a higher 'count'.`